scripts/py_matter_yamltests/matter_yamltests/constraints.py

```diff
@@ -360,6 +360,10 @@ def check_response(self, value, value_type_name) -> bool:
             success = value >= -36028797018963967 and value <= 36028797018963967
         elif self._type == 'nullable_int64s' and type(value) is int:
             success = value >= -9223372036854775807 and value <= 9223372036854775807
+        elif self._type == 'single' and type(value) is float:
+            success = value >= 1.175494351E-38 and value <= 3.402823466E+38
```

Comment:
should negative values also be considered?

Reply:
Oh, that is a very good catch!

Reply:
@andy31415 Let me know if the changes are fine. I have also updated the `chip-tool` adapter since using the new constraint locally make it fails for some cases where the `%g` format specifier returns a value that is not a valid single value.